R/makephyloseqFromTreeSummarizedExperiment.R

```diff
@@ -0,0 +1,113 @@
+#' @export
+setMethod("makePhyloseqFromTreeSummarizedExperiment", signature = c(x = "SummarizedExperiment"),
+          function(x, abund_values = "counts"){
+
+            # Input check
+            # Check object
+            if(!is(x,"TreeSummarizedExperiment")){
+                stop("'x' must be a 'TreeSummarizedExperiment' object")
+            }
+
+            # Check abund_values
+            .check_abund_values(abund_values, x)
+
+            # Get the assay from x
+            otu_table <- assay(x, abund_values)
+            # Convert to otu_table
+            otu_table <- phyloseq::otu_table(otu_table, taxa_are_rows = TRUE)
+
+            # If rowData has information, stores it to tax_table and converts is to phyloseq's tax_table.
+            # Otherwise, tax_table is NULL, and it is converted to phyloseq's tax_table.
+            if(nrow(rowData(x))>0 && ncol(rowData(x))>0){
+                tax_table <- as.matrix(rowData(x))
+                tax_table <- phyloseq::tax_table(tax_table)
+            } else {
+                tax_table <- NULL
+                tax_table <- phyloseq::tax_table(tax_table, errorIfNULL = FALSE)
+            }
+
+            # If colData has information, stores it to sample_data and converts is to phyloseq's sample_data.
+            # Otherwise, sample_data is NULL, and it is converted to phyloseq's sample_data.
+            if(nrow(colData(x))>0 && ncol(colData(x))>0){
+                sample_data <- data.frame(colData(x))
+                sample_data <- phyloseq::sample_data(sample_data)
+            } else {
+                sample_data <- NULL
+                sample_data <- phyloseq::sample_data(sample_data, errorIfNULL = FALSE)
+            }
+
+            # If rowTree has information, stores it to phy_tree and converts is to phyloseq's phy_tree.
+            # Otherwise, phy_tree is NULL, and it is converted to phyloseq's phy_tree.
+            if(!is.null(rowTree(x))){
+                phy_tree <- rowTree(x)
+                phy_tree <- phyloseq::phy_tree(phy_tree)
+            } else {
+                phy_tree <- NULL
+                phy_tree <- phyloseq::phy_tree(phy_tree, errorIfNULL = FALSE)
+            }
+
+            # If referenceSeq has information, stores it to refseq and converts is to phyloseq's refseq.
+            # Otherwise, refseq is NULL, and it is converted to phyloseq's refseq.
+            if(!is.null(referenceSeq(x))){
+                refseq <- referenceSeq(x)
+                refseq <- phyloseq::refseq(refseq)
+            } else {
+                refseq <- NULL
+                refseq <- phyloseq::refseq(refseq, errorIfNULL = FALSE)
+            }
+
+            # Creates a phyloseq object
+            phyloseq <- phyloseq::phyloseq(otu_table,
+                                 tax_table,
+                                 sample_data,
+                                 phy_tree,
+                                 refseq)
```

Comment:
Since the phyloseq constructor dispatches only to `...`, it would be better to construct a list of arguments and call `do.call(phyloseq::phyloseq,args)`

With this call, you don't need to set elements if, they are not present. This remove close to 50% of the code you need to maintain in this function.

Reply:
Code is now using do.call()